core/src/validator.rs

```diff
@@ -533,6 +538,18 @@ impl Validator {
     ) -> Result<Self> {
         let start_time = Instant::now();
 
+        // Initialize the global rayon pool first to ensure the value in config
+        // is honored. Otherwise, some code accessing the global pool could
+        // cause it to get initialized with Rayon's default (not ours)
```

Comment:
I started typing a question to ask why this was moved before I saw this comment. Thanks for adding it 👍